### DISCUSSION TITLE
Not getting InterestCohort Ids from website users

### DISCUSSION THREAD

**Discussion by mohankrishnan** [privacysandbox/privacy-sandbox-dev-support#19] [Replacing cross-site tracking]
Hello,

This is a question connected to joining the FLOC origin trial. If this forum is not the right place to ask this question, kindly re-direct as appropriate. 

We serve destination pages for many leading brands when people engage with their advertisements on media platforms. Couple of days back, we created 3P tokens for the FLOC origin trial and added them to our web pages (which we serve on behalf of the brands - usually served under our domain itself and sometimes under our customer domains). Even after serving a few million pages, we haven't seen even a single interest cohort id coming in (tried in India and Indonesia mainly). I was thinking that we should get interest cohort ids of approximately 5 people in every 1000 chrome page serves (a very rough estimate, I concede). Now I am wondering whether we were understanding / doing this correctly. Here are a couple of sample pages -

1. https://brand-id.shortlyst.com/sunsilk/20103?utm_medium=display&utm_source=youtube&utm_campaign=Sirius%20(Sunsilk)_CN002918_CH2419_BH0468_ID&utm_content=video_bump-rtgstudentsyoungsinglefemale-retail-shopalyst-yellow&utm_term=student&cachekey=displayyoutubeSirius%20(Sunsilk)_CN002918_CH2419_BH0468_IDvideo_bump-rtgstudentsyoungsinglefemale-retail-shopalyst-yellowstudent
2. https://brands-in.shortlyst.com/hamam/214348?utm_medium=display&utm_source=google-ads&utm_campaign=performance_skin_cleansing_hamam_soap_digital_l_amman_new_normal_jun2021_CH1097_BH0235_IN&utm_content=display_Discovery_Custom_Intent&utm_term=broad

Please note that we are getting the interest cohort id when we start chrome with flags - but then for that to work we don't even need the origin trial token. 

It would be really helpful if anyone can take a look and point me to possible issues with the way things are being understood / done by me.

Best regards,
Mohan

**Comment by samdutton** [comment]
Hi Mohan

Thanks for your question, and apologies for the delay in getting back to you. 

> Couple of days back

Are you getting any results now?

Sam

**Comment by mohankrishnan** [comment]
Hi Sam !

Thanks for responding. Unfortunately, do data yet, other  a few - all of them are my team when they start chrome with flags. Does the Meta tag on the look alright to you?

Best,
Mohan

**Reply by samdutton** [reply]
I think the problem might be that the origins registered for the origin trial are different from the origin of the script calling `document.interestCohort()` 

AFAICT the tokens are (successfully) registered for third-party usage, but for the origins `shopalyst.com` and `www.shortlyst.com`, whereas the origin of the script is `brand-id.shortlyst.com`.

Could you try injecting code from `shopalyst.com` or `www.shortlyst.com`, or register for `brand-id.shortlyst.com`?

**Reply by jpchase** [reply]
I agree with Sam that the problem is likely a mismatch in origins. 

You could register for a token like:
Origin: `shortlyst.com`
Matches Subdomains: Yes

The subdomain matching logic uses the entire origin in the token as the suffix, e.g. `example.com` in a token will match `example.com`, `www.example.com`, `subA.example.com`, `subB.example.com`, etc.  But, the matching will not remove `www` in the token. That is, the tokens on this page would match `brand-id.shopalyst.com` or `brand-id.www.shortlyst.com`.

**Reply by mohankrishnan** [reply]
Thanks @samdutton  and @jpchase  ! Have generated 2 3P tokens for shortlyst.com and brands-in.shortlyst.com Will deploy in India a wait for a day to see if this solves the problem. Will keep you posted on what happens. Really appreciate your responses on this. Helps a lot to see that we could get help !

Best regards,
Mohan

**Reply by samdutton** [reply]
Thanks Mohan!

**Reply by mohankrishnan** [reply]
Hi Sam !

Still no data from India, after 24+ hours.  Is there any way for us to get access to a chrome instance that's included in the trial, in which case we could validate the setup? I am a bit puzzled because we had about a Million visitors yesterday from Chrome or Chrome Mobile in India. Was thinking that we should have at least a couple of thousands of people with the FLOC origin trial on..

Best regards
Mohan

**Reply by samdutton** [reply]
Thanks Mohan — I'll take another look. 

@jpchase fyi: on [brands-in.shortlyst.com/hamam/214348](https://brands-in.shortlyst.com/hamam/214348) I can now see tokens (with subdomain matching) for shortlyst.com and brands-in.shortlyst.com (the origin of the script that calls `document.interestCohort()`).

**Reply by jpchase** [reply]
After closer inspection, the problem is that you have registered for third-party tokens, but are providing them as `<meta>` tags, like regular, first-party tokens. (Sorry, I should have caught this in the first go around).

Third-party tokens are intended to be used by embedded content on the page, and so **must** be injected by script, as the [FAQ](https://googlechrome.github.io/OriginTrials/developer-guide.html#18-how-can-i-enable-an-experimental-feature-as-embedded-content-on-different-domains). It doesn't matter which script the interestCohort() API is called from. What matters is *where the token is provided*.

So, your options are:
- Register for first-party tokens (do not select the third-party option), and continue to provide the tokens via `<meta>` tags.
- Inject the existing third-party tokens from a script on the page, which is served from one of the token origins (e.g. brands-in.shortlyst.com).

**Reply by mohankrishnan** [reply]
Thanks a lot @jpchase  ! We missed that too 😬  Anyway, the token change seems to have helped. In the last 48 hours, we got around 14K successful invocations of document.interestCohort() which seems to be fine considering that only 0.5% of chrome instances are supposed to support it. However only 20 had a valid FLOC id and  Version. The rest were all returning empty strings. A few had DNT turned on (about 20), but still the vast majority of others returns an empty string. Is this something that is expected @jpchase  and @samdutton ? Cannot thank you enough for your help on this. Saw that the 3P Cookie end is being pushed out further, but FLOC seems to be a very interesting path to pursue anyway.

Best regards
Mohan

**Reply by samdutton** [reply]
Just getting back onto this — marking as answered.

**Reply by mohankrishnan** [reply]
Thanks @samdutton for your help on this. In the end, we seem to have gotten only very few instances where we got a valid cohort id back. Of about 130K impressions where document.interestCohort() was successfully invoked in India, we got only 130 cohort ids and about 250 cases where do not track was on. All the rest of it returned empty strings. Do you have any views on this?

Where do we go from here? While the Chrome 3P support end seems to have been pushed out, are we not going to go down this path aggressively? We are very interested in the further developments in this space. We meet about 50 M + people on behalf of brand advertisers every month in about 30 countries together and solutions that work beyond 3P cookies are a must have for us.

Thanks again for all the support.

Best regards
Mohan

**Comment by samdutton** [comment]
See jpchase's reply above: https://github.com/GoogleChromeLabs/privacy-sandbox-dev-support/discussions/19#discussioncomment-945582.
